### ISSUE TITLE
Issue: Data Folders don't appear to discover valid data files

### ISSUE THREAD

**Issue by michaeltlombardi** [estruyf/vscode-front-matter#398]
**Describe the bug**

The `frontMatter.data.folders` setting does not appear to work.

**To Reproduce**

1. Create the folder `data/tests` with empty files `foo.json` and `bar.yaml`.
2. In `frontmatter.json`, define a data folder:
   
   ```json
   "frontMatter.data.folders": [
     {
       "id": "Testable",
       "path": "[[workspace]]/data/tests",
       "labelField": "name",
       "schema": {
         "title": "Test Files",
         "type": "object",
         "properties": {
           "name": {
             "title": "Entry Name",
             "type": "string"
           },
           "example": {
             "title": "Example data",
             "type": "string"
           }
         }
       }
     }
   ]
   ```
3. Open the Data view on the dashboard, see that no data files are found.

**Expected behavior**

When I define an entry in `frontMatter.data.folders`, any data files in that folder are found.

**Screenshots**

![Screenshot of the defined data folder in settings, the file structure of the repo showing valid data files in the specified folder, and the data view of the dashboard that found no data files.](https://user-images.githubusercontent.com/14190564/189024763-cb4723cd-0924-413f-9cb8-b610495f5ca8.png)

**Desktop (please complete the following information):**

- VS Code Info:
  - Version: 1.71.0 (system setup)
  - Commit: 784b0177c56c607789f9638da7b6bf3230d47a8c
  - Date: 2022-09-01T07:36:10.600Z
  - Electron: 19.0.12
  - Chromium: 102.0.5005.167
  - Node.js: 16.14.2
  - V8: 10.2.154.15-electron.0
  - OS: Windows_NT x64 10.0.19044
  - Sandboxed: No
- Front Matter: v8.1.3007760-beta

**Additional context**

I suspect that this may be related, like #393, to the path discovery on Windows.

**Comment by estruyf** [comment]
You bug hunter 🤓. Thanks again! It was indeed a parsing issue. You must love how VS Code uses Unix path types but returns Windows file paths. 

The bug has been fixed an available to test out.

**Comment by michaeltlombardi** [comment]
Confirmed fixed!

![Screenshot showing data folders configuration working on Windows](https://user-images.githubusercontent.com/14190564/189115901-303e60c2-287a-466f-98df-27149323cd20.png)

I'm _extremely_ used to cross-platform software having weird edge case bugs on Windows, especially around pathing. 😅 When I worked at Puppet on the Windows team it sometimes felt like that was half of my whole job.

Now that I don't own any code for my day job (mostly) I just try to write up useful bug reports that I would've wanted when I was on the hook for fixing them. 😆
